controllers/machinedeployment_rolling.go

```diff
@@ -167,32 +167,33 @@ func (r *MachineDeploymentReconciler) reconcileOldMachineSets(ctx context.Contex
 
 	log.V(4).Info("Cleaned up unhealthy replicas from old MachineSets", "count", cleanupCount)
 
-	// Scale down old machine sets, need check maxUnavailable to ensure we can scale down
+	// Scale down old machineSets, need check maxUnavailable to ensure we can scale down
 	allMSs = oldMSs
 	allMSs = append(allMSs, newMS)
 	scaledDownCount, err := r.scaleDownOldMachineSetsForRollingUpdate(ctx, allMSs, oldMSs, deployment)
 	if err != nil {
 		return err
 	}
 
-	log.V(4).Info("Scaled down old MachineSets of deployment", "count", scaledDownCount)
+	log.V(4).Info("Scaled down old MachineSets of machineDeployment", "count", scaledDownCount)
 	return nil
 }
 
-// cleanupUnhealthyReplicas will scale down old machine sets with unhealthy replicas, so that all unhealthy replicas will be deleted.
+// cleanupUnhealthyReplicas will scale down old MachineSets with unhealthy replicas, so that all unhealthy replicas will be deleted.
 func (r *MachineDeploymentReconciler) cleanupUnhealthyReplicas(ctx context.Context, oldMSs []*clusterv1.MachineSet, deployment *clusterv1.MachineDeployment, maxCleanupCount int32) ([]*clusterv1.MachineSet, int32, error) {
 	log := ctrl.LoggerFrom(ctx)
 
 	sort.Sort(mdutil.MachineSetsByCreationTimestamp(oldMSs))
 
-	// Safely scale down all old machine sets with unhealthy replicas. Replica set will sort the machines in the order
-	// such that not-ready < ready, unscheduled < scheduled, and pending < running. This ensures that unhealthy replicas will
-	// been deleted first and won't increase unavailability.
+	// Scale down all old MachineSets with any unhealthy replicas. MachineSet will honour Spec.DeletePolicy
```

Comment:
This comment seemed very old left over from pod/replicaSet/Deployment. Please review my update to consolidate with reality.